src/commands/explorer/pinning.ts

```diff
@@ -0,0 +1,43 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { IActionContext } from '@microsoft/vscode-azext-utils';
+import { Memento } from 'vscode';
+import { ext } from '../../extensionVariables';
+import { ResourceGroupsItem } from '../../tree/v2/ResourceGroupsItem';
+import { settingUtils } from '../../utils/settingUtils';
+
+export async function pinTreeItem(_context: IActionContext, treeItem: ResourceGroupsItem): Promise<void> {
+    const memento = getPersistenceMemento();
+    const pinnedItems = memento.get<string[]>('pinnedTreeItems', []);
+
+    pinnedItems.push(treeItem.id); // TODO: need an ID field
+
+    await memento.update('pinnedTreeItems', pinnedItems);

```

Comment:
We need to refresh the tree after pinning or unpinning. However, inside this function we can't tell which tree view the tree item belongs to. 

An easy solution might be to register the pin/unpin commands for each view.

Reply:
Do we expect the IDs to overlap between one view and the other?

Reply:
> Do we expect the IDs to overlap between one view and the other?

I don't think so. But I'm not quite sure what you're getting at here... 😕

Reply:
If I'm not mistaken, as long as the IDs are universally unique, this approach would work for any number of nodes at any level of any number of trees.

Reply:
> I _do_ want to make sure we scope the refreshes to the specific subset we need

Yeah, I think hope we can do that by utilizing `TreeDataProvider.getParent`. However, we'll run into issues if branch tree items want to do pinning, but the branch data provider doesn't implement `getParent`. Though I think that's fine for now since we don't plan on doing that.

Reply:
Sounds good. Didn't we also attach the parent to all of RG's models? i.e. the ones it owns + the wrappers? Or was it just some of them?

Reply:
Only nodes above the wrappers have the parent property. We could restrict the pinning stuff to only those nodes.

Reply:
It sounds like we're gonna do that anyway, at least for now, so this should work 👍